Failure to build

Partial error:
```
gcc -DHAVE_CONFIG_H -I.   -DLIBLOC='"/usr/share/ucblogo/logolib"' -DHELPLOC='"/usr/share/ucblogo/helpfiles"' -DCSLSLOC='"/usr/share/ucblogo/csls"' -DTEMPLOC='"/tmp"'   -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer  -Wno-write-strings -Wno-unused-variable -c -o ucblogo-init.o `test -f 'init.c' || echo './'`init.c
init.c:67:41: error: initialization of ‘NODE * (*)(void)’ {aka ‘struct logo_node * (*)(void)’} from incompatible pointer type ‘NODE * (*)(NODE *)’ {aka ‘struct logo_node * (*)(struct logo_node *)’} [-Wincompatible-pointer-types]
   67 |     {"*", 1, 1, 1, PREFIX_PRIORITY + 3, lmul},
      |                                         ^~~~
init.c:67:41: note: (near initialization for ‘prims[0].prim’)
In file included from init.c:36:
globals.h:143:14: note: ‘lmul’ declared here
  143 | extern NODE *lmul(NODE *);
      |              ^~~~
init.c:68:41: error: initialization of ‘NODE * (*)(void)’ {aka ‘struct logo_node * (*)(void)’} from incompatible pointer type ‘NODE * (*)(NODE *)’ {aka ‘struct logo_node * (*)(struct logo_node *)’} [-Wincompatible-pointer-types]
   68 |     {"+", 1, 1, 1, PREFIX_PRIORITY + 2, ladd},
      |                                         ^~~~
init.c:68:41: note: (near initialization for ‘prims[1].prim’)
globals.h:141:14: note: ‘ladd’ declared here
  141 | extern NODE *ladd(NODE *);
      |              ^~~~
init.c:69:41: error: initialization of ‘NODE * (*)(void)’ {aka ‘struct logo_node * (*)(void)’} from incompatible pointer type ‘NODE * (*)(NODE *)’ {aka ‘struct logo_node * (*)(struct logo_node *)’} [-Wincompatible-pointer-types]
   69 |     {"-", 1, 1, 1, PREFIX_PRIORITY + 2, lsub},
      |                                         ^~~~
init.c:69:41: note: (near initialization for ‘prims[2].prim’)
globals.h:142:14: note: ‘lsub’ declared here
  142 | extern NODE *lsub(NODE *);
      |              ^~~~
init.c:70:42: error: initialization of ‘NODE * (*)(void)’ {aka ‘struct logo_node * (*)(void)’} from incompatible pointer type ‘NODE * (*)(NODE *)’ {aka ‘struct logo_node * (*)(struct logo_node *)’} [-Wincompatible-pointer-types]
   70 |     {"--", 1, 1, 1, PREFIX_PRIORITY + 4, lsub},
      |                                          ^~~~
```
From https://kojipkgs.fedoraproject.org//work/tasks/6987/128156987/build.log

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -16,7 +16,7 @@ jobs:
   build_and_test:
     strategy:
       matrix:
-        wx_version: [ libwxgtk3.0-gtk3-dev, libwxgtk3.2unofficial-dev ]
+        wx_version: [ libwxgtk3.2-dev ]
     name: Build and Test
     runs-on: ubuntu-latest
     timeout-minutes: 5

--- a/coms.c
+++ b/coms.c
@@ -543,7 +543,6 @@ NODE *lshell(NODE *args) {
 #endif
     return(UNBOUND);
 #else
-    extern FILE *popen();
     char cmdbuf[300];
     FILE *strm;
     NODE *head = NIL, *tail = NIL, *this;

--- a/eval.c
+++ b/eval.c
@@ -278,7 +278,7 @@ void reset_args(NODE *old_stack) {
 NODE *bf3(NODE *name) {
     NODE *string = cnv_node_to_strnode(name);
     return make_strnode(getstrptr(string)+3, getstrhead(string),
-			getstrlen(string)-3, nodetype(string), strcpy);
+			getstrlen(string)-3, nodetype(string), strnzcpy);
 }
 
 NODE *deep_copy(NODE *expresn) {

--- a/globals.h
+++ b/globals.h
@@ -50,7 +50,7 @@ extern jmp_buf iblk_buf;
 
 /* logodata.c */
 extern char *strnzcpy(char *, char *, int);
-extern char *word_strnzcpy(char *, NODE *, int);
+extern char *word_strnzcpy(char *, char *, int);
 extern char *noparity_strnzcpy(char *, char *, int);
 extern char *backslashed_strnzcpy(char *, char *, int);
 extern char *mend_strnzcpy(char *, char *, int);
@@ -62,7 +62,7 @@ extern int low_strncmp(char *, char *, int);
 extern int noparity_strncmp(char *, char *, int);
 extern int noparitylow_strncmp(char *, char *, int);
 extern NODE *make_strnode(char *, struct string_block *, int,
-			  NODETYPES, char *(*)());
+			  NODETYPES, char *(*)(char *, char *, int));
 extern void make_runparse(NODE *);
 extern NODE *make_quote(NODE *);
 extern NODE *maybe_quote(NODE *);
@@ -177,7 +177,7 @@ extern NODE *lbeforep(NODE *);
 
 /* intern.c */
 extern NODE *hash_table[HASH_LEN];
-void map_oblist(void (*)());
+void map_oblist(void (*)(NODE *));
 extern NODE *make_instance(NODE *, NODE *);
 extern NODE *intern(NODE *);
 
@@ -617,8 +617,8 @@ extern void wxlPrintPict(void);
 extern void wxlPrintPreviewPict(void);
 extern void wxlPrintText(void);
 extern void wxlPrintPreviewText(void);
-extern NODE*IncreaseFont(void *);
-extern NODE*DecreaseFont(void *);
+extern NODE*IncreaseFont(NODE *);
+extern NODE*DecreaseFont(NODE *);
 extern int turtlePosition_x;
 extern int turtlePosition_y;
 #define SCREEN_WIDTH		1

--- a/init.c
+++ b/init.c
@@ -48,7 +48,7 @@ typedef struct priminfo {
     short defargs;
     short maxargs;
     short priority;
-    NODE *(*prim) ();
+    NODE *(*prim) (NODE *);
 } PRIMTYPE;
 
 NODE *Right_Paren, *Left_Paren, *Redefp, *Caseignoredp, *Erract, *Printdepthlimit,

--- a/intern.c
+++ b/intern.c
@@ -27,7 +27,7 @@
 
 NODE *hash_table[HASH_LEN] = {NIL};
 
-void map_oblist(void (*fcn)()) {
+void map_oblist(void (*fcn)(NODE *)) {
     int i;
     NODE *nd;
 

--- a/logo.h
+++ b/logo.h
@@ -288,6 +288,7 @@ struct string_block {
 #define incstrrefcnt(sh)        (((sh)->str_refcnt)++)
 #define decstrrefcnt(sh)        (--((sh)->str_refcnt))
 
+typedef struct logo_node NODE;
 typedef struct logo_node {
     NODETYPES node_type;
     int my_gen; /* Nodes's Generation */ /*GC*/
@@ -307,7 +308,7 @@ typedef struct logo_node {
 	    FIXNUM nstring_len;
 	} nstring;
 	struct {
-	    struct logo_node * (*nprim_fun) ();
+	    struct logo_node * (*nprim_fun) (NODE *);
 	    short npriority;
 	    short nmin_args;
 	    short ndef_args;

--- a/logodata.c
+++ b/logodata.c
@@ -78,7 +78,8 @@ char *strnzcpy(char *s1, char *s2, int n) {
     return(s1);
 }
 
-char *word_strnzcpy(char *s1, NODE *kludge, int n) {  /* KLUDGE! */
+char *word_strnzcpy(char *s1, char *kludge_char, int n) {  /* KLUDGE! */
+    NODE *kludge = (NODE *)kludge_char;
     char *temp = s1;
 
     while (kludge != NIL) {
@@ -285,7 +286,7 @@ int noparitylow_strncmp(char *s1, char *s2, int n) {
 }
 
 NODE *make_strnode(char *strptr, struct string_block *strhead, int len,
-		   NODETYPES typ, char *(*copy_routine)())
+		   NODETYPES typ, char *(*copy_routine)(char *, char *, int))
 {
     NODE *strnode;
 

--- a/main.c
+++ b/main.c
@@ -196,9 +196,13 @@ void do_keyact(int ch) {
     }
 }
 
-
+#ifdef SIG_TAKES_ARG
+void (*intfuns[])(int) = {0, logo_stop, logo_pause, mouse_down,
+			     delayed_keyact};
+#else
 void (*intfuns[])() = {0, logo_stop, logo_pause, mouse_down,
 			     delayed_keyact};
+#endif
 
 void delayed_int() {
 #ifdef SIG_TAKES_ARG

--- a/term.c
+++ b/term.c
@@ -91,7 +91,7 @@ struct sgttyb tty_cooked, tty_cbreak;
 
 int interactive, tty_charmode;
 
-extern char **environ, *tgoto(), *tgetstr();
+extern char **environ;
 
 char *termcap_ptr;
 

--- a/wrksp.c
+++ b/wrksp.c
@@ -346,11 +346,11 @@ NODE *find_to(NODE *line) {
 		    ((c = *(p-1)) == ' ' || c == '\t' || c == '\n'))
 		return make_strnode(p, getstrhead(line),
 				    getstrlen(line)-(p-lp),
-				    nodetype(line), strcpy);
+				    nodetype(line), strnzcpy);
 	    if (c == '[')
 		return make_strnode(p, getstrhead(line),
 				    strchr(p, ']')-p,
-				    nodetype(line), strcpy);
+				    nodetype(line), strnzcpy);
 	}
 	p++;
     }
@@ -1815,7 +1815,7 @@ NODE *cpdf_newname(NODE *name, NODE*titleline) {
     p2 = p1+strcspn(p1, " \t");
     sprintf(buf, "%.*s%.*s%s",
 	    (int)(p1-titlestr), titlestr, (int)getstrlen(nname), namestr, p2);
-    return make_strnode(buf, NULL, strlen(buf), STRING, strcpy);
+    return make_strnode(buf, NULL, strlen(buf), STRING, strnzcpy);
 }
 
 NODE *lcopydef(NODE *args) {

